src/staging.c

```diff
@@ -359,3 +359,65 @@ int rename_all_files_to_final(struct list *updates)
 
 	return globals.update_count - update_good - update_errs - (globals.update_skip - skip);
 }
+
+enum swupd_code staging_install_all_files(struct list *files, struct manifest *mom)
+{
+	struct list *iter;
+	struct file *file;
+	int ret = 0;
+
+	/*********** rootfs critical section starts ***************************
+NOTE: the next loop calls do_staging() which can remove files, starting a critical section
+which ends after rename_all_files_to_final() succeeds
+	 */
+
+	/* from here onward we're doing real update work modifying "the disk" */
+
+	/* starting at list_head in the filename alpha-sorted updates list
+	 * means node directories are added before leaf files */
+	info("Staging file content\n");
```

Comment:
I like the idea of showing some sort of message here, but I'm uncertain if "Staging file content" is the right message to show to the user. I don't think most users would understand what that means... I even have troubles grasping the concept of staging content.

Reply:
The major problem here is the progress bar. I was planning to add another one here. I'll just move the installing files to the begging of the function and target the progress in a future PR.